### DISCUSSION TITLE
On the fly USDZ generation option disable AR launch button

### DISCUSSION THREAD

**Discussion by NitinJ88** [google/model-viewer#3076] [Q&A]
### Description
I want to use on the fly USDZ  generation option in model viewer by not providing ios-src attribute. It disabled the AR launch button on model-viewer if ios-src attribute is not provided . On some comment on twitter is says it disable AR button till it generates USDZ but for me button did not come up after waited ~15min. 

#### Live Demo

https://glitch.com/edit/#!/model-viewer

![model_with_ios_src](https://user-images.githubusercontent.com/86343802/147100611-544ca196-8fe8-435c-9118-58cc64cf59dd.JPG)
![model_without_ios_src](https://user-images.githubusercontent.com/86343802/147100615-bcf7c930-06ee-439b-962b-c0bb86c61758.JPG)
![with_ios_src](https://user-images.githubusercontent.com/86343802/147100618-8513a3c0-f891-4c3b-9e8f-a74ff707ccdd.JPG)
![without_ios_src](https://user-images.githubusercontent.com/86343802/147100622-4ba97c19-7db5-4d12-89fd-62c5cfc9ebac.JPG)

### Version

- model-viewer: v1.9.1

### Browser Affected

- [* ] Chrome, version: 96.0.4664.104

### OS

- [ *] Android 10

### AR

- [*] WebXR
- [*] QuickLook

**Comment by elalish** [comment]
It looks like you forgot to specify  `quick-look` as  an `ar-mode`: https://modelviewer.dev/docs/index.html#entrydocs-augmentedreality-attributes-arModes

**Reply by NitinJ88** [reply]
I have set the ar-mode to "quick-look" still the AR launch button is disable. Please check the screen shot for ar-mode I used.

**Reply by grapix** [reply]
Make sure, that you initialize the modelviewer with an src attribute, even it is empty.

**Reply by NitinJ88** [reply]
@grapix : Thanks for you reply. Setting src attribute to empty by default fixed the issue.
